packages/flutter/lib/src/services/keyboard_key.dart

```diff
@@ -0,0 +1,2543 @@
+  /// “o” for [keyLabel], but would return “ö” for [RawKeyEvent.character].
+  ///
+  /// {@macro flutter.services.RawKeyEventData.keyLabel}
+  final String keyLabel;
+
+  @override
+  int get hashCode => keyId.hashCode;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    final LogicalKeyboardKey typedOther = other;
+    return keyId == typedOther.keyId;
+  }
+
+  /// Finds the [LogicalKeyboardKey] that matches the given ID.
+  static LogicalKeyboardKey findKeyByKeyId(int keyId) => _knownLogicalKeys[keyId];
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('keyId', '0x${keyId.toRadixString(16).padLeft(8, '0')}', showName: true));
+    properties.add(StringProperty('keyLabel', keyLabel, showName: true));
+    properties.add(StringProperty('debugName', debugName, showName: true, defaultValue: null));
+  }
+
+  /// Returns true if the given label represents a Unicode control character.
+  ///
+  /// Examples of control characters are characters like "U+000A LINE FEED (LF)"
+  /// or "U+001B ESCAPE (ESC)".
+  ///
+  /// See <https://en.wikipedia.org/wiki/Unicode_control_characters> for more
+  /// information.
+  ///
+  /// Used by [RawKeyEvent] subclasses to help construct IDs.
+  static bool isControlCharacter(String label) {
+    if (label.length > 1) {
+      return false;
+    }
+    final int codeUnit = label.codeUnitAt(0);
+    return (codeUnit <= 0x1f && codeUnit >= 0x00) || (codeUnit >= 0x7f && codeUnit <= 0x9f);
+  }
+
+  /// Returns true if the [keyId] of this object is one that is autogenerated by
+  /// Flutter.
+  ///
+  /// Autogenerated key IDs are generated in response to platform key codes
+  /// which Flutter doesn't recognize, and their IDs shouldn't be used in a
+  /// persistent way.
+  ///
+  /// Autogenerated IDs should be a rare occurrence: Flutter supports most keys.
+  ///
+  /// Keys that generate Unicode characters (even if unknown to Flutter) will
+  /// not return true for `isAutogenerated`, since they will be assigned a
+  /// Unicode-based code that will remain stable.
+  ///
+  /// If Flutter adds support for a previously unsupported key code, the ID it
+  /// reports will change, but the ID will remain stable on the platform it is
+  /// produced on until Flutter adds support for recognizing it.
+  ///
+  /// So, hypothetically, if Android added a new key code of 0xffff,
+  /// representing a new "do what I mean" key, then the autogenerated code would
+  /// be 0x1020000ffff, but once Flutter added the "doWhatIMean" key to the
+  /// definitions below, the new code would be 0x0020000ffff for all platforms
+  /// that had a "do what I mean" key from then on.
+  bool get isAutogenerated => (keyId & autogeneratedMask) != 0;
+
+  /// Mask for the 32-bit value portion of the key code. This is used by
```

Comment:
nit (here and below): First paragraph should be a one sentence summary.

Reply:
Fixed.